### ISSUE TITLE
Товары в наличии пропадают в каталоге

### ISSUE THREAD

**Issue by Juicew** [wpcraft-ru/wooms#302]
После обновления плагина до 7.3
Товары которые в моём складе имеются в наличии по 5-10 штук, они предаются корректно на сайт, количество в наличии передаётся, но в карточке товара отображается как скрытый.

Подскажите в чем может быть причина ?

**Comment by antonr600** [comment]
> После обновления плагина до 7.3
> Товары которые в моём складе имеются в наличии по 5-10 штук, они предаются корректно на сайт, количество в наличии передаётся, но в карточке товара отображается как скрытый.
> 
> Подскажите в чем может быть причина ?

Так же и у меня.
Вчера пропали из каталога товары. Есть подозрение, что просто последние несколько дней не скрывались правильно товары на 7.3, а после ручной синхронизации и вовсе пропали многие. _До этого была версия 7.1. После перешел на 7.3._

Вчера начал заново ручную синхронизацию, в итоге товары начали скрываться и появляться с нуля в каталоге, после чего их оказалось из потенциальных 700 около 50 видимых в каталоге.
После тестов с различными версиями уже уменьшились до 3-8 товаров.
На версии 7.4 также всё, при откате на 7.3 и 7.2, все тоже самое.

7.1 ещё не побывал.

Зависимость от темы не заметил, переключался на дефолтные вордпресса, не помогло.

**Comment by antonr600** [comment]
_**Временное решение:**_
Вручную менять статус видимости в каталоге товаров, которые в наличии.
Возможно установить плагин для массового редактирования товаров.

![Inked2020-06-03_16-20-14_LI](https://user-images.githubusercontent.com/66369773/83642149-22d1e280-a5b7-11ea-997b-bdd37f37e230.jpg)

**Comment by antonr600** [comment]
Поменял массово видимость товаров в каталоге и поиске вручную.
Использовал плагин для массового редактирования товаров woocommerce advanced bulk edit.

После обновления статусов видимости, дефолтный виджет категорий в shop сайдбаре считал их пустыми, кроме одной, которая была 1 до изменений видимости. Сначала отключил скрытие пустых категорий, удалил пустые. Но в итоге висела категория по тематике нового года, что мешало.

Решил поэксперементировать. Повторно вручную запустил синк. Товары все остались, НГ категория скрылась. Обратно включил в сайдбаре шоп виджета категорий скрытие пустых каталогов.

Ещё буду смотреть как будет себя вести при появлении новых товаров.
На данный момент установлены последние версии Wordpress и Woo. WP в режиме мультисайта. Тема Shopkeeper последней версии 2.9.26. WooMS также последний актуальный 7.4.

**Comment by Juicew** [comment]
Я тоже пробовал и без плагинов дополнительных и на новой теме, рекомендованную автором плагина.
И кстати что как то выборочно скрывает товары. Потому что я добавлял после этого новые товары и обновлял остатки в моем складе и всё четко проходит на сайт.

**Comment by uptimizt** [comment]
@Juicew @antonr600 
пробовали повторный синк?

иногда бывает проблема на вариативных.
помогает 2-3 синка.

иначе - можно отправить мне доступ к вам сайтам через hd@wpcraft.ru

я постараюсь оперативно посмотреть на реальном конфиге.

так как на нашем тест сайте эта проблема не проявляется.

**Comment by Juicew** [comment]
Да, пробовал вручную запускать синк раз 5-6 с отключением доп функций плагина. Не помогло. Пришлю доступы на почту

**Comment by uptimizt** [comment]
добавил логгер остатков в мета поля для поиска проблемы
![image](https://user-images.githubusercontent.com/1852897/83950171-4a33e400-a831-11ea-82d9-52620d70f06c.png)
